frappe/utils/print_format.py

```diff
@@ -88,6 +89,16 @@ def read_multi_pdf(output):
 @frappe.whitelist()
 def download_pdf(doctype, name, format=None, doc=None, no_letterhead=0):
 	html = frappe.get_print(doctype, name, format, doc=doc, no_letterhead=no_letterhead)
+	# fetch file auto name format from doctype.
+	file_auto_name = frappe.get_meta(doctype).file_auto_name
+
+	if not doc:
+		doc = frappe.get_doc(doctype, name)
+
+	if file_auto_name:
+		# based on type of format used set_name_form_naming_option return result.
+		name = set_name_from_naming_options(file_auto_name, doc)
```

Comment:
```suggestion
	if file_auto_name and not doc:
		# based on type of format used set_name_form_naming_option return result.
		doc = frappe.get_doc(doctype, name)
		name = set_name_from_naming_options(file_auto_name, doc)
```

Reply:
```suggestion
	if file_auto_name and not doc: 
```
this will fail if Request for a quotation as function call from RFQ send doc args

Reply:
I mean it will bypass this condition in rfq